paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/FlinkSourceBuilder.java

```diff
@@ -294,6 +295,9 @@ public DataStream<RowData> build() {
         }
 
         if (sourceBounded) {
+            if (conf.get(CoreOptions.SCAN_BATCH_PLAN_MOVE_FROM_JM_TO_TM_ENABLED)) {
```

Comment:
If users configured this option and their job is in streaming mode, can this optimization still take effect? Would MonitorSource achieve a similar optimization in streaming mode?